src/charm.py

```diff
@@ -369,8 +369,21 @@ def _on_update_status(self, event):
         if self.unit.is_leader():
             self._handle_reconfigure(event)
 
-        # update the units status based on it's replica set config.
-        self.unit.status = build_unit_status(self.mongodb_config, self._unit_ip(self.unit))
+        # update the units status based on it's replica set config and backup status. An error in
+        # the status of MongoDB takes precedence over pbm status.
+        mongodb_status = build_unit_status(self.mongodb_config, self._unit_ip(self.unit))
+        pbm_status = self.backups._get_pbm_status()
+        if (
+            isinstance(mongodb_status, WaitingStatus)
+            or isinstance(mongodb_status, BlockedStatus)
```

Comment:
maybe, not sure
```suggestion
            not isinstance(mongodb_status, ActiveStatus)
```

Reply:
this is also a solid suggestion thank you Mykola